central/auth/datastore/singleton.go

```diff
@@ -15,7 +24,19 @@ var (
 // Singleton provides a singleton auth machine to machine DataStore.
 func Singleton() DataStore {
 	once.Do(func() {
-		ds = New(pgStore.New(globaldb.GetPostgres()))
+		set := m2m.TokenExchangerSetSingleton(roleDataStore.Singleton(), jwt.IssuerFactorySingleton())
+		ds = New(pgStore.New(globaldb.GetPostgres()), set)
+
+		// On initialization of the store, list all existing configs and fill the set.
+		ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS), sac.ResourceScopeKeys(resources.Access)))
+		configs, err := ds.ListAuthM2MConfigs(ctx)
+		utils.Should(err)
+		for _, config := range configs {
+			exchanger, err := set.NewTokenExchangerFromConfig(ctx, config)
+			utils.Should(err)
+			utils.Should(set.UpsertTokenExchanger(exchanger, config.GetId()))
```

Comment:
It should instantiate the next exchanger if the current one failed. `utils.Should` doesn't break the loop.
```suggestion
			exchanger, err := set.NewTokenExchangerFromConfig(ctx, config)
			utils.Should(err)
			if err != nil {
				continue
			}
			utils.Should(set.UpsertTokenExchanger(exchanger, config.GetId()))
```

Also, `utils.Should` starts its logged message with `Unexpected Error: `. I don't think it's that unexpected here: creation of exchanger may fail, e.g. due to http request not going through. OTOH, the advantage of `utils.Should` is that it's hard to miss in development builds. I would be fine to keep it in case all `err`-s here provide meaningful message.